pkg/services/opsgenie.go

```diff
@@ -15,8 +16,13 @@ import (
 )
 
 type OpsgenieOptions struct {
-	ApiUrl  string            `json:"apiUrl"`
-	ApiKeys map[string]string `json:"apiKeys"`
+	ApiUrl              string            `json:"apiUrl"`
+	ApiKeys             map[string]string `json:"apiKeys"`
+	InsecureSkipVerify  bool              `json:"insecureSkipVerify"`
+	MaxIdleConns        int               `json:"maxIdleConns"`
+	MaxIdleConnsPerHost int               `json:"maxIdleConnsPerHost"`
+	MaxConnsPerHost     int               `json:"maxConnsPerHost"`
+	IdleConnTimeout     time.Duration     `json:"idleConnTimeout"`
```

Comment:
We can't use time.Duration for parsing time values - we will need to use a string and then parse the actual `time.Duration` by doing something like this 

```golang
	idleConnTimeout, err := time.ParseDuration(s.opts.IdleConnTimeout)
	if err != nil {
		return errors.Wrap(err, "failed to parse idle connection timeout")
	}
```

and then passing in `idleConnTimeout` to the `NewTransport` function.